safe_transaction_service/tokens/services/price_service.py

```diff
@@ -248,6 +249,14 @@ def get_cached_usd_values(self, token_addresses: Sequence[ChecksumAddress]) -> I
             logger.warning('Cannot get network ether price', exc_info=True)
             eth_price = 0
 
+        # If the token is the native one (NULL_ADDRESS) we can return the eth_price immediately
+        if len(token_addresses) == 1 and token_addresses[0] == NULL_ADDRESS:
```

Comment:
I don't think this is ok. I think it shouldn't be important the number of `token_addresses` received. Maybe yielding `1` in `get_cached_token_eth_values` if `token == NULL_ADDRESS`?

Reply:
Also, we can reduce the cache for `get_eth_usd_price`

Reply:
> I don't think this is ok. I think it shouldn't be important the number of `token_addresses` received. Maybe yielding `1` in `get_cached_token_eth_values` if `token == NULL_ADDRESS`?

Good point! Much better now. Actually `get_cached_token_eth_values` was already doing that so it was a matter of adjusting the condition on the view to not require the native coin to be in the Token table. Also adjusted the test for it.

> Also, we can reduce the cache for get_eth_usd_price

I would do this as part of a different PR if we agree on a new value 👍